addon/components/frost-object-details.js

```diff
@@ -1,5 +1,77 @@
 import Ember from 'ember'
+import layout from '../templates/components/frost-object-details'
+import _ from 'lodash'
 
 export default Ember.Component.extend({
-  classNames: ['frost-object-details']
+
+  _routing: Ember.inject.service('-routing'),
+
+  layout: layout,
+
+  classNames: ['frost-object-details'],
+
+  parentRouteName: Ember.computed('_routing.currentRouteName', function () {
+    let currentRouteName = this.get('_routing.currentRouteName')
+    if (_.includes(currentRouteName, '.views.')) {
+      return currentRouteName.substring(0, currentRouteName.indexOf('.views.'))
+    } else if (_.includes(currentRouteName, '.related.')) {
+      return currentRouteName.substring(0, currentRouteName.indexOf('.related.'))
+    }
+  }),
+
+  viewRoutes: Ember.computed('_routing', 'parentRouteName', function () {
+    let availableRoutes = Object.keys(this.get('_routing.router').router.recognizer.names)
+    return _.chain(availableRoutes)
+      .filter((route) => {
+        return _.includes(route, `${this.get('parentRouteName')}.views.`) && !(_.includes(route, 'loading') ||
+          _.includes(route, 'error') ||
+          _.includes(route, 'index'))
+      })
+      .map((viewRoute) => {
+        const { getOwner } = Ember
+        let label = ''
+        if (getOwner(this).lookup(`route:${viewRoute}`)) {
+          label = getOwner(this).lookup(`route:${viewRoute}`).get('detailsLabel')
+        }
+        return {
+          alias: label || viewRoute.substring(`${this.get('parentRouteName')}.views.`.length),
+          route: viewRoute
+        }
+      })
+      .value()
+  }),
+
+  relatedRoutes: Ember.computed('_routing', 'parentRouteName', function () {
+    let availableRoutes = Object.keys(this.get('_routing.router').router.recognizer.names)
+    return _.chain(availableRoutes)
+      .filter((route) => {
+        return _.includes(route, `${this.get('parentRouteName')}.related.`) && !(_.includes(route, 'loading') ||
+          _.includes(route, 'error') ||
+          _.includes(route, 'index'))
+      })
+      .map((relatedRoute) => {
+        const { getOwner } = Ember
+        let label = ''
+        let labelSvgPath = ''
+        if (getOwner(this).lookup(`route:${relatedRoute}`)) {
```

Comment:
Save the initial lookup to a `let` so that the lookup only needs to be done once